### DISCUSSION TITLE
A function which returns intent and message at same time

### DISCUSSION THREAD

**Discussion by OnlinePage** [iparsw/custom-intent#8] [General]
Hi, so we have two different function for getting message and the tag of the message.
It would be really nice if we can have one single function as well which returns both. Since if we have to get intent we have use `request_tag` which make one single iteration and also for the message we have `request_response` which will make one iteration.

So in due cases where both are required then, the two iteration are time costly.

My suggestion is to have a function which return both of them at single iteration as tuple

Also if my last suggestion about NER is considered then it would be really nice to have it combined all along with the function which return all three at same time.

Thanks.

**Comment by OnlinePage** [comment]
@iparsw any thoughts on this?

**Reply by iparsw** [reply]
Hi sorry i just saw your post that’s a good idea i will add it as soon as i can and I didn’t still read your last post. Any suggestions as what the function should be named?

**Reply by OnlinePage** [reply]
ahh! alirght, so the name can be like `returnMessageWithIntent()` that is how I have made the function on me end, which returns tuple. Here is what I have done:

`
def request_tagWithrespone(message):
    return assistant.request_response(message),assistant.request_tag(message)
`
but this way we are making two iterations and extra ms of time is being consumed.

**Reply by iparsw** [reply]
I will add it to the next update but it’s exams month here and im a student so it might take some time

**Reply by OnlinePage** [reply]
Yeah yea, I understand these bullshit exams , me student as well.😐😔
